docs/babel.config.js

```diff
@@ -21,6 +21,7 @@ const alias = {
   '@mui/docs': '../node_modules/@mui/monorepo/packages/mui-docs/src',
   '@mui/markdown': '../node_modules/@mui/monorepo/docs/packages/markdown',
   '@mui/monorepo': '../node_modules/@mui/monorepo',
+  '@mui/joy': '../node_modules/@mui/monorepo/packages/mui-joy/src',
```

Comment:
`_document` in monorepo started to use `@mui/joy` in https://github.com/mui/material-ui/pull/32576

Initially, I wanted to add `@mui/joy` dependency to our `docs/package.json`, but it can get out of sync with monorepo quickly. I think using alias is a better.